firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/LaunchView/LaunchScreenViewModelTests.swift

```diff
@@ -135,6 +135,337 @@ final class LaunchScreenViewModelTests: XCTestCase {
         XCTAssertTrue(updatedValue)
     }
 
+    // MARK: - Multiple Launch Types Tests
+
+    func testLaunchType_termsOfServiceAndIntro_sequence() {
+        FxNimbus.shared.features.tosFeature.with(initializer: { _, _ in
+            TosFeature(status: true)
+        })
+
+        let subject = createSubject()
+        subject.delegate = delegate
+        subject.startLoading(appVersion: "112.0")
+
+        XCTAssertEqual(delegate.launchBrowserCalled, 0)
+        XCTAssertEqual(delegate.finishedLoadingLaunchOrderCalled, 1)
+        XCTAssertEqual(subject.launchOrder.count, 2)
+
+        subject.loadNextLaunchType()
+        guard case .termsOfService = delegate.savedLaunchType else {
+            XCTFail("Expected terms of service, but was \(String(describing: delegate.savedLaunchType))")
+            return
+        }
```

Comment:
nit: since we do this in multiple places, maybe be can add a helper method 

```
private func assertSavedLaunchType(_ expected: LaunchType, file: StaticString = #file, line: UInt = #line) {
    guard let saved = delegate.savedLaunchType else {
        XCTFail("Expected \(expected) but savedLaunchType was nil", file: file, line: line)
        return
    }
    // If LaunchType is Equatable:
    XCTAssertEqual(saved, expected, "Expected \(expected) but was \(saved)", file: file, line: line)
}
```